### PR TITLE
feat(trading): single deposit flow

### LINKED ISSUE #6702: Custom Squid-SDK powered deposit flow (and removal of widget)
## The Chore

Create a new intergrated Squid SDK deposit flow which supports a single flow from beginning to end and utilises the source and target chain definition to determine the optimal deposit path and execute it using the Arbitrum bridge and Squid SDK.

This flow would replace the current same chain / cross chain widget implementation.

## Questions

- [x] Follow basic sketch order shown below
- [ ] When user sets the chain we should trigger a WAGMI network switch
- [ ] Ensure that if you try to deposit VEGA we default the source and show a warning
- [ ] Ensure we show the various info that squid SDK provides regarding fees / slippage etc etc
- [ ] Ensure that the approval amount behaves as we want it to
- [ ] If the source chain = target chain and asset is the same just use normal deposit flow ... no SDK
- [ ] If the source chain = target chain but asset is different then use the SDK
- [ ] If the source chain <> target chain then obviously use the SDK

## Additional details / background info

image

### PR DESCRIPTION
**Pull request by mattrussell36**
# Related issues 🔗

Closes #6702

# Description ℹ️

Uses squid routers SDK to have a single deposit form

# Demo 📺

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
   },
   "private": true,
   "dependencies": {
-    "@0xsquid/staking-widget": "^1.0.6",
+    "@0xsquid/sdk": "^2.8.23",
```
Comment (dexturr) [inline package.json:24]: revert this

File: apps/trading/components/deposit-container/deposit-container.tsx [hunk 2]
```diff
@@ -67,284 +27,31 @@ export const DepositContainer = ({
   const asset = assets?.find((a) => a.id === initialAssetId);
 
   if (loading) {
-    return null;
+    return <div>Loading</div>;
```
Comment (asiaznik) [inline apps/trading/components/deposit-container/deposit-container.tsx:30]: `t` - translation missing

File: apps/trading/components/deposit-container/deposit-form.tsx [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+        <SubmitButton
+          isSwap={isSwap}
+          isFetchingRoute={isFetching}
+          isExecuting={status === 'pending'}
+          estimate={routeData?.route.estimate}
+        />
+        {status === 'pending' && (
+          <div className="flex flex-col gap-2 items-center text-xs mt-2">
+            <p>
+              Swap and deposit in progress.{' '}
```
Comment (asiaznik) [inline apps/trading/components/deposit-container/deposit-form.tsx:216]: `t` - translation missing

File: apps/trading/components/deposit-container/deposit-form.tsx [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+                </a>
+              )}
+            </p>
+          </div>
+        )}
+        {status === 'error' && (
+          <div className="flex flex-col items-center gap-2 mt-2">
+            <div className="flex items-center gap-1 text-xs">
+              <VegaIcon name={VegaIconNames.CROSS} />
+              <p>Deposit failed</p>
```
Comment (asiaznik) [inline apps/trading/components/deposit-container/deposit-form.tsx:234]: `t`

File: apps/trading/components/deposit-container/deposit-form.tsx [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+            </div>
+            <p>{squidExecuteError.message}</p>
+          </div>
+        )}
+        {status === 'success' && (
+          <div className="flex flex-col items-center gap-2 mt-2">
+            <div className="flex items-center gap-1 text-xs">
+              <VegaIcon name={VegaIconNames.TICK} />
+              <p>Swap and deposit complete</p>
```
Comment (asiaznik) [inline apps/trading/components/deposit-container/deposit-form.tsx:243]: `t`

File: apps/trading/components/deposit-container/swap-info.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        );
+      })}
+      {Object.entries(feeGroups).map(([key, group]) => {
+        const fees = group.map((f) => {
+          return toBigNum(f.amount, f.token.decimals);
+        });
+        const total = BigNumber.sum.apply(null, fees);
+        return (
+          <div key={key} className="grid grid-cols-2">
+            <dt className="text-surface-1-fg-muted">Estimated fees</dt>
```
Comment (asiaznik) [inline apps/trading/components/deposit-container/swap-info.tsx:60]: `t`

File: apps/trading/components/deposit-container/swap-info.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        </dd>
+      </div>
+      {Object.entries(gasGroups).map(([key, group]) => {
+        const fees = group.map((f) => {
+          return toBigNum(f.amount, f.token.decimals);
+        });
+        const total = BigNumber.sum.apply(null, fees);
+        return (
+          <div key={key} className="grid grid-cols-2">
+            <dt className="text-surface-1-fg-muted">Gas costs</dt>
```
Comment (asiaznik) [inline apps/trading/components/deposit-container/swap-info.tsx:46]: `t`

File: apps/trading/components/deposit-container/use-squid-route.ts [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+              value: '0',
+              callData: depositCallData,
+              payload: {
+                tokenAddress: toAsset.source.contractAddress as `0x${string}`,
+                inputPos: 1,
+              },
+              estimatedGas: '50000',
+            },
+          ],
+          description: 'Swap and deposit',
```
Comment (asiaznik) [inline apps/trading/components/deposit-container/use-squid-route.ts:181]: we could use `t` here as well

File: apps/trading/components/deposit-container/use-squid-route.ts [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+          abi: getErc20Abi({
+            address: toAsset.source.contractAddress as `0x${string}`,
+          }),
+          functionName: 'approve',
+          args: [bridgeAddress, BigInt(fromAmount)],
+        });
+
+        // NOTE that the squid receiver deposit contract takes a 4th argument which is the recovery
+        // address if squid is unable to fulfill the swap
+        depositCallData = encodeFunctionData({
```
Comment (asiaznik) [inline apps/trading/components/deposit-container/use-squid-route.ts:114]: suggestion: we have `encodeDepositData` and `encodeApproveData` on the contracts class that could be reused here

File: apps/trading/components/asset-option/asset-option.tsx [hunk 10]
```diff
@@ -24,9 +28,68 @@ export const AssetOption = ({
...
+
+  if (!data) return null;
+
+  return <>{addDecimalsFormatNumber(data.value.toString(), data.decimals)}</>;
+};
+
+const TokenBalance = (props: AssetBalanceProps) => {
+  const { address } = useAccount();
+
+  const tokenAddress = props.address as `0x${string}`;
```
Comment (asiaznik) [inline apps/trading/components/asset-option/asset-option.tsx:67]: suggestion: `0x${string}` is used in couple of places how about making it a common type e.g. `HexString`

File: apps/trading/components/deposit-container/use-squid-route.ts (hunk 9 above)
Comment (mattrussell36) [reply apps/trading/components/deposit-container/use-squid-route.ts:114]: I decided to switch to using wagmi/viem helpers rather that our wrappers that still use ethers.

**Review by asiaznik — APPROVED** [review]

**Pull request merged**